hystrix-contrib/hystrix-metrics-event-stream/src/main/java/com/netflix/hystrix/contrib/sample/stream/HystrixSampleSseServlet.java

```diff
@@ -146,12 +146,16 @@ public void onError(Throwable e) {
                             @Override
                             public void onNext(String sampleDataAsString) {
                                 if (sampleDataAsString != null) {
-                                    writer.print("data: " + sampleDataAsString + "\n\n");
-                                    // explicitly check for client disconnect - PrintWriter does not throw exceptions
-                                    if (writer.checkError()) {
+                                    try {
+                                        writer.print("data: " + sampleDataAsString + "\n\n");
+                                        // explicitly check for client disconnect - PrintWriter does not throw exceptions
+                                        if (writer.checkError()) {
+                                            moreDataWillBeSent.set(false);
+                                        }
+                                        writer.flush();
+                                    } catch (Throwable t) {
```

Comment:
Catching `Throwable` is generally a bad idea, these catches should probably be `Exception`